### PR TITLE
UCT/RDMACM: make the rdma_resolve_addr/route timeout configurable.

### PR DESCRIPTION
**Pull request by alinask**
@yosefe please review.

### CODE REVIEW

File: src/uct/base/uct_cm.c [hunk 1]
```diff
@@ -20,6 +20,10 @@ ucs_config_field_t uct_cm_config_table[] = {
    "Allow using an address that is already in use.",
    ucs_offsetof(uct_cm_config_t, allow_addr_inuse), UCS_CONFIG_TYPE_BOOL},
 
+  {"ADDR_ROUTE_RESOLVE_TIMEOUT", "10000ms",
```
Comment (yosefe) [inline src/uct/base/uct_cm.c]: "TIMEOUT", "10s"

File: src/uct/base/uct_cm.c [hunk 2]
```diff
@@ -20,6 +20,10 @@ ucs_config_field_t uct_cm_config_table[] = {
...
+   "Time to wait for address/route resolution to complete",
```
Comment (yosefe) [inline src/uct/base/uct_cm.c]: "Timeout for RDMA address and route resolve operations",

File: src/uct/base/uct_cm.c [hunk 3]
```diff
@@ -239,6 +243,7 @@ UCS_CLASS_INIT_FUNC(uct_cm_t, uct_cm_ops_t* ops, uct_iface_ops_t* iface_ops,
     self->iface.progress_flags    = 0;
 
     self->config.allow_addr_inuse = config->allow_addr_inuse;
+    self->config.resolve_timeout  = config->resolve_timeout;
```
Comment (yosefe) [inline src/uct/base/uct_cm.c]: resolve_timeout  ->timeout

File: src/uct/ib/rdmacm/rdmacm_cm.c [hunk 4]
```diff
@@ -88,7 +89,8 @@ static void uct_rdmacm_cm_handle_event_addr_resolved(struct rdma_cm_event *event
               uct_rdmacm_cm_ep_str(cep, ep_str, UCT_RDMACM_EP_STRING_LEN),
               event->id);
 
-    if (rdma_resolve_route(event->id, 1000 /* TODO */)) {
+    if (rdma_resolve_route(event->id, UCS_MSEC_PER_SEC *
+                           cm->super.config.resolve_timeout)) {
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_cm.c]: static inline int uct_rdmacm_get_timeout(cm) 
{
  return  UCS_MSEC_PER_SEC * cm->super.config.resolve_timeout
}

reuse in uct_rdamcm_cm_ep_client_init

**Comment by alinask** [comment]
@yosefe updated.

**Review by yosefe — APPROVED** [review]

**Pull request merged**